Backpack/Map/Classes/BPKMapAnnotationView.m

```diff
@@ -0,0 +1,162 @@
+#import "BPKMapAnnotation.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKMapAnnotationView()
+@property(nonatomic, nullable) BPKMapAnnotation *bpk_annotation;
+@property(nonatomic, nullable, strong) UIView *dotView;
+@property(nonatomic, nullable, strong) BPKMapAnnotationViewCalloutView *calloutView;
+@property(nonatomic, assign) BOOL hasBeenSelected;
+@end
+
+@implementation BPKMapAnnotationView
+
+- (instancetype)initWithAnnotation:(id<MKAnnotation> _Nullable)annotation reuseIdentifier:(NSString * _Nullable)reuseIdentifier {
+    self = [super initWithAnnotation:annotation reuseIdentifier:reuseIdentifier];
+
+    if (self) {
+        [self setupAppearance];
+    }
+
+    return self;
+}
+
+- (instancetype)init {
+    self = [super init];
+
+    if (self) {
+        [self setupAppearance];
+    }
+
+    return self;
+}
+
+-(BPKMapAnnotation *_Nullable)bpk_annotation {
+    if(self.annotation == nil) {
+        return nil;
+    }
+
+    if(![self.annotation isKindOfClass:BPKMapAnnotation.class]) {
+        return nil;
+    }
+
+    BPKMapAnnotation *annotation = (BPKMapAnnotation *)self.annotation;
+    return annotation;
+}
+
+-(void)setupAppearance {
+    [self updateImage];
+
+    self.dotView = [[UIView alloc] initWithFrame:CGRectZero];
+    [self addSubview:self.dotView];
+    self.dotView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:@[
+        // These constraints use magic numbers to ensure that the dot appears in the correct position on the map.
+        // The UIKit PinAnnotationView isn't centered by default.
+        [self.dotView.centerXAnchor constraintEqualToAnchor:self.leadingAnchor constant:8],
+        [self.dotView.centerYAnchor constraintEqualToAnchor:self.bottomAnchor constant:-2],
+        [self.dotView.widthAnchor constraintEqualToConstant:BPKSpacingLg],
+        [self.dotView.heightAnchor constraintEqualToConstant:BPKSpacingLg]
+    ]];
+    self.dotView.backgroundColor = BPKColor.skyBlue;
+    self.dotView.layer.borderColor = BPKColor.white.CGColor;
+    self.dotView.layer.borderWidth = BPKSpacingSm;
+    self.dotView.layer.cornerRadius = BPKSpacingLg/2;
+    [self updateCallout];
+}
+
+-(void)createCallout {
+    if (self.dotView == nil) {
+        return;
+    }
+
+    self.calloutView = [[BPKMapAnnotationViewCalloutView alloc] initWithText:self.annotation.title];
+    self.calloutView.enabled = self.enabled;
+    self.calloutView.selected = self.selected;
+    [self addSubview:self.calloutView];
+    self.calloutView.translatesAutoresizingMaskIntoConstraints = NO;
+    [NSLayoutConstraint activateConstraints:@[
+        [self.calloutView.centerXAnchor constraintEqualToAnchor:self.dotView.centerXAnchor],
+        [self.calloutView.bottomAnchor constraintEqualToAnchor:self.dotView.topAnchor constant:2]
+    ]];
+}
+
+-(void)destroyCallout {
+    [self.calloutView removeFromSuperview];
+    self.calloutView = nil;
+}
+
+-(void)updateImage {
+    self.image = nil;
```

Comment:
🤔

Reply:
The underlying class shows an image of a pin. We can set the image to be nil as we want to use a subview instead of an image.

We are also required to set the image when the InterfaceStyle changes, otherwise it will be reverted by the underlying component.

![simulator_screenshot_9DD7B9DA-8628-4BBE-890A-53BBE090A2BD](https://user-images.githubusercontent.com/30267516/107045549-8a32d880-67bd-11eb-8043-f697e935f44d.png)